yt/visualization/tests/test_plotwindow.py

```diff
@@ -804,3 +808,61 @@ def test_dispatch_plot_classes():
     assert isinstance(p2, OffAxisProjectionPlot)
     assert isinstance(s1, AxisAlignedSlicePlot)
     assert isinstance(s2, OffAxisSlicePlot)
+
+
+def test_swap_axes():
+
+    # note: this tests the swap_axes functionality separately from the
+    # ATTR_ARGS framework because swap_axes currently is not valid with
+    # callbacks.
```

Comment:
I don't know what's the ATTR_ARGS framework. Also, I thought this _was_ compatible with callbacks. It'd be a shame to introduce this functionality without that compatibility, so what's your approach ? how much work do you think it would take and are you positive that it could be done before the 4.1 release ?

Reply:
latest commit gets this working with most callbacks.